evm/src/cpu/decode.rs

```diff
@@ -147,6 +147,19 @@ pub fn eval_packed_generic<P: PackedField>(
         * (opcode - P::Scalar::from_canonical_usize(0xfc_usize))
         * lv.op.m_op_general;
     yield_constr.constraint(m_op_constr);
+
+    // Manually check lv.op.jumpdest_keccak_general.
+    // KECCAK_GENERAL is a kernel-only instruction, but not JUMPDEST.
+    // JUMPDEST is differentiated from KECCAK_GENERAL by its second bit set to 1.
+    yield_constr.constraint(
+        (P::ONES - kernel_mode) * lv.op.jumpdest_keccak_general * (P::ONES - lv.opcode_bits[1]),
+    );
+
+    // Check the JUMPDEST and KERNEL_GENERAL opcodes.
+    let jumpdest_keccak_general_constr = (opcode - P::Scalar::from_canonical_usize(0x21_usize))
+        * (opcode - P::Scalar::from_canonical_usize(0x5b_usize))
```

Comment:
nit: we don't need to specify `_usize` here, the compiler will infer the type.
Additionally for clarity, I think it could be worth defining the two opcodes similarly to how it's done below in the circuit version, i.e.:

```rust
    let jumpdest_opcode = P::Scalar::from_canonical_usize(0x5b_usize);
    let keccak_general_opcode = P::Scalar::from_canonical_usize(0x21_usize);
    let jumpdest_keccak_general_constr = (opcode - keccak_general_opcode)
        * (opcode - jumpdest_opcode) * lv.op.jumpdest_keccak_general;
    yield_constr.constraint(jumpdest_keccak_general_constr);
```